sops.go

```diff
@@ -189,6 +190,53 @@ func (branch TreeBranch) Set(path []interface{}, value interface{}) TreeBranch {
 	return set(branch, path, value).(TreeBranch)
 }
 
+func unset(branch interface{}, path []interface{}) (interface{}, error) {
+	switch branch := branch.(type) {
+	case TreeBranch:
+		for i, item := range branch {
+			if item.Key == path[0] {
+				if len(path) == 1 {
+					branch = slices.Delete(branch, i, i+1)
+				} else {
+					v, err := unset(item.Value, path[1:])
+					if err != nil {
+						return nil, err
+					}
+					branch[i].Value = v
+				}
+				return branch, nil
+			}
+		}
+		return nil, fmt.Errorf("Key not found: %s", path[0])
+	case []interface{}:
+		position := path[0].(int)
+		if position >= len(branch) {
+			return nil, fmt.Errorf("Index %d out of bounds (maximum: %d)", position, len(branch))
```

Comment:
Similar here. (Though obviously it's only idempotent if you delete the last element from an array.)